src/text-pango.c

```diff
@@ -156,9 +156,11 @@ gdip_process_string (gchar *text, int length, int removeAccelerators, int trimSp
 		iter = iter2;
 	}
 	/* always ensure that at least one space is measured */
-	if (!nonws && trimSpace) {
-		g_string_append_c (res, ' ');
-		j++;
+	if (!nonws && trimSpace && length > 0) {
+		iter = text;
+		iter2 = g_utf8_next_char (iter);
+		g_string_append_len (res, iter, iter2 - iter);
+		j += iter2 - iter;
```

Comment:
Would it be possible to make simple unit test for this? Just so we can compare the behavior on Windows and Linux.

Reply:
Should be, measure a string with, for example, an em-space (the issue I had). I'll have a go tomorrow.

Reply:
Well, I hope I've sorted out counting the characters.

But I did a few checks on Windows (via .Net) as to the behaviour there, and it is weird. To me, it looks like unless there is a line ending, the only whitespace removed from the end are actual spaces. If there is a line ending, all trailing whitespace (considering tab as non-whitespace!) is removed. Furthermore, there is no leaving one space / character behind (gives a width of 0).

Reply:
I've just added a test case for this (aded2ce). I have also added fixes for the reported number of characters fitted (in two parts, one issue was counting the characters removed (109fe41), the other was in counting the characters in the layout (5b6db10)).